homeassistant/components/myfitnesspal/sensor.py

```diff
@@ -0,0 +1,68 @@
+"""Support for Myfitnesspal totals as sensors."""
+from datetime import date
+from homeassistant.const import MASS_GRAMS
+from homeassistant.helpers.entity import Entity
+
+# REQUIREMENTS = ['myfitnesspal==1.13.3']
+TOTALS = ["sodium", "carbohydrates", "calories", "fat", "sugar", "protein"]
+
+
+def setup_platform(hass, config, add_devices, discovery_info=None):
+    """Set the sensor platform."""
+    dev = []
+    for resource in TOTALS:
+
+        dev.append(
+            MyFitnessPalSensor(resource, config)
+        )
+
+    add_devices(dev, True)
+
+
+class MyFitnessPalSensor(Entity):
+    """Representation of a Sensor."""
+
+    ICON = 'mdi:barley'
+
+    def __init__(self, resource, config):
+        """Initialize the sensor."""
+        self._state = None
+        self._user = config['user']
```

Comment:
use home assistant constants for username and password

Reply:
These are the login for the service not ya credentials? Would that work this way?

Reply:
the const.py file has standard constants to use. instead of user and pass we need to use username and password

Reply:
Believe this is covered now.